web-console/src/components/auto-form.tsx

```diff
@@ -100,6 +103,15 @@ export class AutoForm<T> extends React.Component<AutoFormProps<T>, AutoFormState
     </HTMLSelect>
   }
 
+  private renderJSONInput(field: Field): JSX.Element {
+    const { model, onChange, JSONErrors, updateJSONErrors } = this.props;
+    return <JSONInput
+      value={(model as any)[field.name]}
+      onChange={(e: any) => onChange(Object.assign({}, model, { [field.name]: parseStringToJSON(e)}))}
```

Comment:
it would be neater if `JSONInput` has the same value type as the `onChange` type, meaning move `parseStringToJSON` into JSON input